### ISSUE TITLE
"warning: left shift count >= width of type" in test suite

### ISSUE THREAD

**Issue by jserv** [DLTcollab/sse2neon#600]
During running `make`, there are various warnings:
```
tests/impl.cpp:6001:41: warning: left shift count >= width of type [-Wshift-count-overflow]
 6001 |     d##IDX[6] = (IDX & ~15) ? 0 : _a[6] << IDX; \
tests/common.h:291:5: note: in expansion of macro ‘TEST_IMPL’
  291 |     TEST_IMPL(63)
      |     ^~~~~~~~~
tests/impl.cpp:6008:5: note: in expansion of macro ‘IMM_64_ITER’
 6008 |     IMM_64_ITER
      |     ^~~~~~~~~~~
tests/impl.cpp:6002:41: warning: left shift count >= width of type [-Wshift-count-overflow]
 6002 |     d##IDX[7] = (IDX & ~15) ? 0 : _a[7] << IDX; \
tests/common.h:291:5: note: in expansion of macro ‘TEST_IMPL’
  291 |     TEST_IMPL(63)
      |     ^~~~~~~~~
tests/impl.cpp:6008:5: note: in expansion of macro ‘IMM_64_ITER’
 6008 |     IMM_64_ITER
      |     ^~~~~~~~~~~
```

Tested with g++-9 and g++-10.

**Comment by jserv** [comment]
Close via commit [6a2d069](https://github.com/DLTcollab/sse2neon/commit/6a2d069bc57a54b48869f98ec2eca9e4ff9a08b8)

**Comment by jserv** [comment]
@howjmay, MSVC failed to run with the proposed changes. See https://github.com/DLTcollab/sse2neon/actions/runs/5398011257/jobs/9804146367#step:4:533

**Comment by howjmay** [comment]
Copy that!

**Comment by howjmay** [comment]
Closed by #602. Thanks for @pbo-linaro and @anthony-linaro for help
